src/jenkins.py

```diff
@@ -347,4 +375,129 @@ def add_agent_node(
     except jenkinsapi.custom_exceptions.AlreadyExists:
         pass
     except jenkinsapi.custom_exceptions.JenkinsAPIException as exc:
+        logger.error("Failed to add agent node, %s", exc)
         raise JenkinsError("Failed to add agent node.") from exc
+
+
+def _get_major_minor_version(version: str) -> str:
+    """Extract the major.minor version from semantic version string.
+
+    Args:
+        version: The semantic version.
+
+    Returns:
+        The version without patch version, i.e. <major>.<minor>
+    """
+    return ".".join(version.split(".")[0:2])
+
+
+def _fetch_versions_from_rss() -> typing.Iterable[str]:
+    """Fetch and extract Jenkins versions from the stable RSS feed.
+
+    Returns:
+        The jenkins versions from the RSS feed.
+
+    Raises:
+        JenkinsNetworkError: if there was an error fetching the RSS feed.
+        ValidationError: if an invalid RSS feed was received.
+    """
+    try:
+        res = requests.get(RSS_FEED_URL, timeout=30)
+        res.raise_for_status()
+    except (
+        requests.exceptions.ConnectionError,
+        requests.exceptions.Timeout,
+        requests.exceptions.HTTPError,
+    ) as exc:
+        logger.error("Failed to fetch latest RSS feed, %s", exc)
+        raise JenkinsNetworkError("Failed to fetch RSS feed.") from exc
+
+    try:
+        # jenkins xml is a trusted source, hence it can be parsed using stdlib
+        xml_tree = ElementTree.fromstring(res.content)  # nosec
+    except ElementTree.ParseError as exc:
+        logger.error("Invalid RSS feed, %s", exc)
+        raise ValidationError("Invalid RSS feed.") from exc
+
+    items = xml_tree.findall("./channel/item")
+    # mypy doesn't understand that None type is not possible.
+    titles = (
+        item.find("title").text  # type: ignore
+        for item in items
+        if item.find("title") is not None and item.find("title").text is not None  # type: ignore
+    )
+    versions = (title.removeprefix("Jenkins ") for title in titles)  # type: ignore
+    return versions
+
+
+def get_latest_patch_version(current_version: str) -> str:
+    """Get the latest lts patch version matching with the current version.
+
+    Args:
+        current_version: Current LTS semantic version.
+
+    Returns:
+        The latest patched version available.
+
+    Raises:
+        JenkinsNetworkError: if there was an error fetching the LTS RSS feed.
+        ValidationError: if the RSS feed contains no matching LTS version.
+    """
+    try:
+        versions = _fetch_versions_from_rss()
+    except JenkinsNetworkError as exc:
+        raise JenkinsNetworkError("Failed to fetch LTS versions from RSS feed.") from exc
+
+    maj_min_version = _get_major_minor_version(current_version)
+    for version in versions:
+        if version.startswith(maj_min_version):
+            return version
```

Comment:
This will return the first one matching, versions isn't sorted, are we sure the RSS feed will always return a sorted list ?
Wouldn't it be safer to get all the one that match and then return the highest patch version ?
I'm wondering if using a semver library could help as semver could be more that X.Y.Z

Reply:
From my understanding, the RSS feed was always sorted from latest to oldest, but yes, it would be beneficial to be safer, thanks for the suggestion!

Reply:
I've tried refactoring with all versions using `semver` library. However, there were some caveats that I led me think that at this moment, it could be better to do string parsing.
1. Jenkins versions are not semantic - i.e. 2.387, we can parse this to `semver.Version` type. However, when converting it back to `str` (needed in most cases), it becomes `2.387.0`, without an option to omit the last patch version 0. This meant that whenever I need to convert the `semver.Version` back to `str`, the custom `version_to_string` method would need to follow everywhere. I'm not too sure if that would be beneficial.
2. The stable RSS feed would return  versions of format in x.y.z consistently. I think that we can rely on this not changing for a while.